feature-libs/customer-ticketing/core/facade/customer-ticketing.service.ts

```diff
@@ -1,6 +1,73 @@
 import { Injectable } from '@angular/core';
+import {
+  Query,
+  QueryNotifier,
+  QueryService,
+  RoutingService,
+  UserIdService,
+} from '@spartacus/core';
+import {
+  CustomerTicketingFacade,
+  GetTicketQueryReloadEvent,
+  GetTicketQueryResetEvent,
+  TicketDetails,
+} from '@spartacus/customer-ticketing/root';
+import { combineLatest, Observable } from 'rxjs';
+import { map, switchMap, take } from 'rxjs/operators';
+import { CustomerTicketingConnector } from '../connectors';
 
-@Injectable({
-  providedIn: 'root',
-})
-export class CustomerTicketingService {}
+@Injectable()
+export class CustomerTicketingService implements CustomerTicketingFacade {
+  /**
+   * Returns the reload events for the getTicket query.
+   */
+  protected getTicketQueryReloadEvents(): QueryNotifier[] {
+    return [GetTicketQueryReloadEvent];
+  }
+  /**
+   * Returns the reset events for the getTicket query.
+   */
+  protected getTicketQueryResetEvents(): QueryNotifier[] {
+    return [GetTicketQueryResetEvent];
+  }
+
+  protected getTicketQuery$: Query<TicketDetails | undefined> =
+    this.queryService.create<TicketDetails | undefined>(
+      () =>
+        this.checkTicketPreConditions().pipe(
+          switchMap(([customerId, ticketId]) =>
+            this.customerTicketingConnector.getTicket(customerId, ticketId)
+          )
+        ),
+      {
+        reloadOn: this.getTicketQueryReloadEvents(),
+        resetOn: this.getTicketQueryResetEvents(),
+      }
+    );
+
+  constructor(
+    protected queryService: QueryService,
+    protected userIdService: UserIdService,
+    protected customerTicketingConnector: CustomerTicketingConnector,
+    protected routingService: RoutingService
+  ) {}
+
+  protected checkTicketPreConditions(): Observable<[string, string]> {
+    return combineLatest([
+      this.userIdService.getUserId(),
+      this.routingService.getParams(),
```

Comment:
instead of having it re-emit for different params if ever that becomes the case, then maybe we can compose this differently as follows:

```
 combineLatest([
this.userIdService.getUserId(),
this.routingService..getRouterState().pipe(
      map((routingData) => routingData.state.params.orderCode),
      distinctUntilChanged()
)
])
```

What do you think?

Reply:
yes we do this. but, why can't we use the getParams method instead?

Reply:
like this:
```
combineLatest([
      this.userIdService.getUserId(),
      this.routingService.getParams().pipe(
        map((params) => params.ticketCode),
        distinctUntilChanged()
      ),
    ])
```

Reply:
that's fine too